extensions/aws/s3/writer/src/main/java/org/eclipse/dataspaceconnector/writer/s3/S3BucketWriter.java

```diff
@@ -0,0 +1,80 @@
+package org.eclipse.dataspaceconnector.writer.s3;
+
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.eclipse.dataspaceconnector.provision.aws.AwsTemporarySecretToken;
+import org.eclipse.dataspaceconnector.schema.s3.S3BucketSchema;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+import org.eclipse.dataspaceconnector.spi.types.TypeManager;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataAddress;
+import org.eclipse.dataspaceconnector.transfer.inline.spi.DataWriter;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+public class S3BucketWriter implements DataWriter {
+
+    private final RetryPolicy<Object> retryPolicy;
+    private final Monitor monitor;
+    private final TypeManager typeManager;
+
+    public S3BucketWriter(Monitor monitor, TypeManager typeManager, RetryPolicy<Object> retryPolicy) {
+        this.monitor = monitor;
+        this.typeManager = typeManager;
+        this.retryPolicy = retryPolicy;
+    }
+
+    @Override
+    public boolean canHandle(String type) {
+        return "AmazonS3".equals(type);
+    }
+
+    @Override
+    public Result<Void> write(DataAddress destination, String name, byte[] data, String secretToken) {
+        var bucketName = destination.getProperty(S3BucketSchema.BUCKET_NAME);
+        var region = destination.getProperty(S3BucketSchema.REGION);
+        var awsSecretToken = typeManager.readValue(secretToken, AwsTemporarySecretToken.class);
+
+        try (S3Client s3 = S3Client.builder()
+                .credentialsProvider(buildCredentialsProvider(awsSecretToken))
+                .region(Region.of(region))
+                .build()) {
+            return transferData(s3, bucketName, data, name);
+        } catch (S3Exception ex) {
+            monitor.severe("Data request: transfer failed!");
+            return Result.failure("Failed to transfer data");
+        }
+    }
+
+    private Result<Void> transferData(S3Client client, String bucketName, byte[] data, String name) {
```

Comment:
this method can be inlined, since it does not a lot and the exception is already caught from the `write` method